### PR TITLE
CoordinateParser should handle NumState.invalid first

### PR DESCRIPTION
**Pull request by wieslawsoltes**
#### Reference Issue

**e-polyline-004.svg**
```SVG
<svg id="svg1" viewBox="0 0 200 200" xmlns="http://www.w3.org/2000/svg">
    <title>Stop processing on invalid data</title>
    <desc>`text` and everything after it should be ignored</desc>

    <!-- should be covered -->
    <path id="path1" d="M 20 40 L 160 180 L 30 150" fill="none" stroke="red"/>

    <polyline id="polyline1" points="20 40 160 180 30 150 text" fill="none" stroke="green"/>

    <!-- image frame -->
    <rect id="frame" x="1" y="1" width="198" height="198" fill="none" stroke="black"/>
</svg>
```

**e-polygon-004.svg**
```SVG
<svg id="svg1" viewBox="0 0 200 200" xmlns="http://www.w3.org/2000/svg">
    <title>Stop processing on invalid data</title>
    <desc>`text` and everything after it should be ignored</desc>

    <!-- should be covered -->
    <path id="path1" d="M 20 40 L 160 180 L 30 150" fill="red" stroke="red"/>

    <polygon id="polygon1" points="20 40 160 180 30 150 text" fill="green" stroke="green"/>

    <!-- image frame -->
    <rect id="frame" x="1" y="1" width="198" height="198" fill="none" stroke="black"/>
</svg>
```

### CODE REVIEW

**Comment by wieslawsoltes** [comment]
> Can you add a release note entry for the fix? Also, a test would be nice (e.g. test image + PNG).

Added.

File: Tests/W3CTestSuite/svg/__pull_request-640-02.svg [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (H1Gdev) [inline Tests/W3CTestSuite/svg/__pull_request-640-02.svg]: Please apply EditorConfig settings.

File: Tests/W3CTestSuite/svg/__pull_request-640-01.svg [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (H1Gdev) [inline Tests/W3CTestSuite/svg/__pull_request-640-01.svg]: Please apply EditorConfig settings.

File: Tests/W3CTestSuite/PassingTests.txt [hunk 3]
```diff
@@ -279,6 +279,8 @@ __pull_request-551-01.svg
 __pull_request-564-01.svg
 __pull_request-564-02.svg
 __pull_request-575-01.svg
+__pull_request-640-01.svg
```
Comment (H1Gdev) [inline Tests/W3CTestSuite/PassingTests.txt]: Please add tests in Tests/Svg.UnitTests/PassingTests.csv, too.

File: Tests/W3CTestSuite/svg/__pull_request-640-02.svg (hunk 1 above)
Comment (wieslawsoltes) [reply Tests/W3CTestSuite/svg/__pull_request-640-02.svg]: Fixed

File: Tests/W3CTestSuite/svg/__pull_request-640-01.svg (hunk 2 above)
Comment (wieslawsoltes) [reply Tests/W3CTestSuite/svg/__pull_request-640-01.svg]: Fixed

File: Tests/W3CTestSuite/PassingTests.txt (hunk 3 above)
Comment (wieslawsoltes) [reply Tests/W3CTestSuite/PassingTests.txt]: Fixed

**Comment by wieslawsoltes** [comment]
Fixed review comments

**Comment by mrbean-bremen** [comment]
Thanks!

**Pull request merged**